sys/pm_callouts.c

```diff
@@ -468,28 +445,13 @@ void send_icmp_blocked_packet(portmaster_packet_info* packetInfo, void* original
         // Calculate checksum for the icmp packet
         calc_ipv6_checksum(icmpPacket, packetLength, TRUE);
 
-        // Create the net buffer list for the new ICMP packet.
-        status = wrap_packet_data_in_nb(icmpPacket, packetLength, &injectNBL);
-        if (!NT_SUCCESS(status)) {
-            ERR("ICMPv6 Port unreachable -> wrap_packet_data_in_nb failed: 0x%x", status);
-            portmaster_free(icmpPacket);
-            return;
-        }
-
-        if(packetInfo->direction == 1 || is_ipv6_loopback(packetInfo->remoteIP) || useLocalHost) {
-            // InjectNetworkReceive does not work for loopback packet but send works!?
-            status = FwpsInjectNetworkSendAsync(inject_out6_handle, 0, 0, UNSPECIFIED_COMPARTMENT_ID,
-                    injectNBL, free_after_inject,
-                    icmpPacket);
+        // Reverse diraction and inject packet
+        reverseDirection = packetInfo->direction == 1 ? 0 : 1;
```

Comment:
Should the direction regard the `useLocalHost` flag?

Reply:
This is checked in inject_packet function